django/core/cache/backends/filebased.py

```diff
@@ -35,14 +35,13 @@ def add(self, key, value, timeout=DEFAULT_TIMEOUT, version=None):
 
     def get(self, key, default=None, version=None):
         fname = self._key_to_file(key, version)
-        if os.path.exists(fname):
-            try:
-                with io.open(fname, 'rb') as f:
-                    if not self._is_expired(f):
-                        return pickle.loads(zlib.decompress(f.read()))
-            except IOError as e:
-                if e.errno == errno.ENOENT:
-                    pass  # Cache file was removed after the exists check
+        try:
+            with io.open(fname, 'rb') as f:
+                if not self._is_expired(f):
+                    return pickle.loads(zlib.decompress(f.read()))
+        except IOError as e:
+            if e.errno == errno.ENOENT:
+                pass  # Cache file does not exists
```

Comment:
> What's the point of checking the errno here?

Yes, you're right. It was carried over from the old implementation, but doesn't look right.

I'm not sure the original authors exact intention, but I'm guessing this should be doing the same as the other file existence checks. Would we ever expect an errno other than `ENOENT` here under normal conditions?

```
        except IOError as e:
            if e.errno != errno.ENOENT:
                raise
```